lldb fails to build on `CommandObjectTraceStartIntelPT.cpp:81:21: error: calling a protected constructor of class 'lldb_private::Status'`

On trunk, Debian amd64:

```
FAILED: tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/CommandObjectTraceStartIntelPT.cpp.o
/opt/sccache//sccache /build/source/build-llvm/bin/clang++ -DHAVE_ROUND -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/source/Plugins/Trace/intel-pt -I/build/source/lldb/source/Plugins/Trace/intel-pt -I/build/source/lldb/include -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/include -I/build/source/build-llvm/tools/clang/stage2-bins/include -I/build/source/llvm/include -I/usr/include/python3.12 -I/build/source/clang/include -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/../clang/include -I/build/source/lldb/source -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/source -isystem /usr/include/libxml2 -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -ffile-prefix-map=/build/source/build-llvm/tools/clang/stage2-bins=../../../../ -ffile-prefix-map=/build/source/= -no-canonical-prefixes -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-deprecated-register -Wno-vla-extension -O2 -DNDEBUG -g1 -std=c++17  -fno-exceptions -funwind-tables -MD -MT tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/CommandObjectTraceStartIntelPT.cpp.o -MF tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/CommandObjectTraceStartIntelPT.cpp.o.d -o tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/CommandObjectTraceStartIntelPT.cpp.o -c /build/source/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp
/build/source/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp:81:21: error: calling a protected constructor of class 'lldb_private::Status'
   81 |     result.SetError(Status(std::move(err)));
      |                     ^
/build/source/lldb/include/lldb/Utility/Status.h:173:3: note: declared protected here
  173 |   Status(llvm::Error error);
      |   ^
/build/source/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp:167:21: error: calling a protected constructor of class 'lldb_private::Status'
  167 |     result.SetError(Status(std::move(err)));
      |                     ^
/build/source/lldb/include/lldb/Utility/Status.h:173:3: note: declared protected here
  173 |   Status(llvm::Error error);
      |   ^
2 errors generated.
```

Log:
https://llvm-jenkins.debian.net/job/llvm-toolchain-binaries/architecture=amd64,distribution=unstable,label=amd64/2010/console

--- a/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp
+++ b/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp
@@ -78,7 +78,7 @@ bool CommandObjectThreadTraceStartIntelPT::DoExecuteOnThreads(
     llvm::ArrayRef<lldb::tid_t> tids) {
   if (Error err = m_trace.Start(tids, m_options.m_ipt_trace_size,
                                 m_options.m_enable_tsc, m_options.m_psb_period))
-    result.SetError(Status(std::move(err)));
+    result.SetError(std::move(err));
   else
     result.SetStatus(eReturnStatusSuccessFinishResult);
 
@@ -164,7 +164,7 @@ void CommandObjectProcessTraceStartIntelPT::DoExecute(
           m_options.m_ipt_trace_size, m_options.m_process_buffer_size_limit,
           m_options.m_enable_tsc, m_options.m_psb_period,
           m_options.m_per_cpu_tracing, m_options.m_disable_cgroup_filtering))
-    result.SetError(Status(std::move(err)));
+    result.SetError(std::move(err));
   else
     result.SetStatus(eReturnStatusSuccessFinishResult);
 }

